archeryutils/targets.py

```diff
@@ -147,35 +202,61 @@ def __repr__(self) -> str:
 
     def __eq__(self, other: object) -> bool:
         """Check equality of Targets based on parameters."""
-        if isinstance(other, Target):
-            return self._parameters() == other._parameters()
-        return NotImplemented
+        if not isinstance(other, Target):
+            return NotImplemented
+
+        if self.scoring_system == other.scoring_system == "Custom":
+            return (
+                self._face_spec == other._face_spec
+                and self._parameters() == other._parameters()
+            )
+        return self._parameters() == other._parameters()
 
     def _parameters(self):
         """Shortcut to get all target parameters as a tuple for comparison."""
         return (
-            self.scoring_system,
-            self.diameter,
-            self.native_diameter_unit,
-            self.distance,
-            self.native_dist_unit,
+            self._scoring_system,
+            self._diameter,
+            self._native_diameter_unit,
+            self._distance,
+            self._native_dist_unit,
             self.indoor,
         )
```

Comment:
Hmm, yeah this could be a good idea, its theoretically redudant work (for a non-custom target, the face spec doesn't encode any new information that isnt already contained in the paramters above), but there could be something to be said for having it in there anyway to reduce branching and special casing in __eq__, and also any future functionality that might want a definitive representation of the parameters making up a target.

Reply:
Yes, I think so.
And if we are worried about the performance hit of generating a facespec I would suggest we might be using the wrong language ;)